apicapi/apic_client.py

```diff
@@ -1097,16 +1097,19 @@ def renew(self, mo, *params):
 
     @contextlib.contextmanager
     def transaction(self, transaction=None, ph=None, top_send=False):
-        if not transaction:
-            transaction = Transaction(self, top_send=top_send)
-            yield transaction
-            if transaction.root:
-                result = transaction.commit()
-                if ph is not None:
-                    ph.append(result)
-        else:
-            # Only the top owner will commit the transaction
-            yield transaction
+        try:
```

Comment:
I think we need a different solution here. It looks like the unit test is getting the StopIteration due to the way the UT has been constructed - in this case, it looks like the mock uses a list of side effects, which triggers the StopIteration exception. Rather than have our code work around this UT issue, I think we should instead figure how to handle this exception with a change to the UT code. That might require a bit more digging to find the right fix, but it would be better than catching the StopIteration here.